src/data_model/mod.rs

```diff
@@ -1,4 +1,5 @@
 mod device;
 mod event;
+pub mod protocol;
```

Comment:
I do not think protocol is a part of the data model, maybe make a new module

Reply:
Fixed